fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/search/test/JDBCSearchNearTest.java

```diff
@@ -0,0 +1,296 @@
+import com.ibm.fhir.persistence.MultiResourceResult;
+import com.ibm.fhir.persistence.SingleResourceResult;
+import com.ibm.fhir.persistence.context.FHIRPersistenceContext;
+import com.ibm.fhir.persistence.context.FHIRPersistenceContextFactory;
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+import com.ibm.fhir.persistence.jdbc.impl.FHIRPersistenceJDBCImpl;
+import com.ibm.fhir.persistence.jdbc.test.util.DerbyInitializer;
+import com.ibm.fhir.search.context.FHIRSearchContext;
+import com.ibm.fhir.search.util.SearchUtil;
+
+/**
+ * <a href="https://www.hl7.org/fhir/r4/location.html#positional">FHIR
+ * Specification: Positional Searching for Location Resource</a>
+ * <br>
+ * Original LONG/LAT
+ * 
+ * <pre>
+ *  "position": {
+ *      "longitude": -83.6945691,
+ *      "latitude": 42.25475478,
+ *      "altitude": 0
+ * }
+ * </pre>
+ */
+public class JDBCSearchNearTest {
+    private Properties testProps;
+
+    protected Location savedResource;
+
+    protected static FHIRPersistence persistence = null;
+
+    @BeforeClass
+    public void startup() throws Exception {
+        InputStream inputStream = JDBCSearchNearTest.class.getResourceAsStream("/near.unitTest.properties");
+        LogManager.getLogManager().readConfiguration(inputStream);
+
+        FHIRConfiguration.setConfigHome("../fhir-persistence/target/test-classes");
+        FHIRRequestContext.get().setTenantId("default");
+
+        testProps = TestUtil.readTestProperties("test.jdbc.properties");
+
+        DerbyInitializer derbyInit;
+        String dbDriverName = this.testProps.getProperty("dbDriverName");
+        if (dbDriverName != null && dbDriverName.contains("derby")) {
+            derbyInit = new DerbyInitializer(this.testProps);
+            derbyInit.bootstrapDb();
+        }
+
+        savedResource = TestUtil.readExampleResource("json/spec/location-example.json");
+
+        persistence   = new FHIRPersistenceJDBCImpl(this.testProps);
+
+        SingleResourceResult<Location> result =
+                persistence.create(FHIRPersistenceContextFactory.createPersistenceContext(null), savedResource);
+        assertTrue(result.isSuccess());
+
+    }
+
+    public MultiResourceResult<Resource> runQueryTest(String searchParamCode, String queryValue) throws Exception {
+        Map<String, List<String>> queryParms = new HashMap<String, List<String>>(1);
+        if (searchParamCode != null && queryValue != null) {
+            queryParms.put(searchParamCode, Collections.singletonList(queryValue));
+        }
+
+        FHIRSearchContext ctx = SearchUtil.parseQueryParameters(Location.class, queryParms, true);
+        FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, ctx);
+        MultiResourceResult<Resource> result = persistence.search(persistenceContext, Location.class);
+        return result;
+    }
+
+    @AfterClass
+    public void teardown() throws Exception {
```

Comment:
should we also delete the saved resource(s)?

Reply:
sure ... that's no problem to add

Reply:
done